### ISSUE TITLE
Rotating file sink crashes app on Windows XP

### ISSUE THREAD

**Issue by o-mdr** [gabime/spdlog#389]
I have ported loging for an app from BOOST logging to spdlog. During testing I've discovered that `rotating_file_sink_mt` throws exception and crashes my app. I've narrowed it down to this line:

```
auto foo = std::make_shared<spdlog::sinks::rotating_file_sink_mt>("C:\\a", "log", 123, 3);
```

If I wrap it in a `try-catch`

```
try
{
    auto foo = std::make_shared<spdlog::sinks::rotating_file_sink_mt>("C:\\a", "log", 123, 3);
}
catch (std::exception& e)
{	
	std::cout << e.what(); // prints Failed getting file size from fd: Invalid argument
}
```
If I remove `try-catch` I get this pop-up
![image](https://cloud.githubusercontent.com/assets/3082246/24003816/6a57af80-0a5c-11e7-8ef5-084c52359f16.png)

The app is compiled as 32 bit and runs on Win XP version: x64 5.2 Build 3790.svr03_sp2_rtm.070216-1710 SP2. Both multi-threaded and single-threaded rotating sinks are affected. I've tested other file sinks (simple and daily) - they work fine. Also works fine on Windows 10.

**Comment by o-mdr** [comment]
Narrowed it down to `os.h`

```
//Return file size according to open FILE* object
inline size_t filesize(FILE *f)
{
...
#else //windows 32 bits
    struct _stat st;
    if (_fstat(fd, &st) == 0)  // <============== HERE  the result isn't 0
        return st.st_size;
#endif
     ...
    throw spdlog_ex("Failed getting file size from fd", errno);
}
```

**Comment by gabime** [comment]
Thanks for reporting. Could you send a pull to fix/workaround it ?

**Comment by o-mdr** [comment]
@gabime I will try to figure it out. It seems to be specific to only Windows XP and `stat` (and its variants) and v14_xp compiler. 

I'll try keep changes to the minimum and only modify `inline size_t filesize(FILE *f)` to use another function to get size on Win XP.

**Comment by gabime** [comment]
sounds good

**Comment by o-mdr** [comment]
I came up with a solution 

```
// On Windows platrorm only x86 and x64
// If _fsstat fail, e.g. running on Win XP using v14_xp toolset
// try resolving the size of the file by fseek/ftell before throwing an exception
// https://github.com/gabime/spdlog/issues/389	
size_t original_pos = ftell(f); // remember the original position
if (fseek(f, 0, SEEK_END) == 0) // seek to end
{
	size_t size = ftell(f);
	fseek(f, original_pos, SEEK_SET); // rewind to where we were
	return size;
}
```

But it looks like someone has already re-wrote this part to use `_filelength` which works fine on Win XP.

I am using the released version 0.12.0 and the bug is still there. However, I've decided not to commit the code, as the current `develop` branch is working fine on Win XP - so no need to overwrite the changes (and my solution uses more lines).  

You can mark this as resolved for the upcoming release.
